tests/Api/Utils/HeadersBuilder.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Tests\Api\Utils;
+
+use Lexik\Bundle\JWTAuthenticationBundle\Services\JWTTokenManagerInterface;
+use MonorepoBuilderPrefix202310\Webmozart\Assert\Assert;
+use Sylius\Component\User\Model\UserInterface;
+use Sylius\Component\User\Repository\UserRepositoryInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\Security\Core\User\UserInterface as SymfonyUserInterface;
+
+class HeadersBuilder
+{
+    /**
+     * @var array <int, string>
+     */
+    private array $headers = [];
+
+    public function __construct(private ContainerInterface $container)
+    {
+    }
+    public function withJsonContentType(): self
+    {
+        $this->headers['CONTENT_TYPE'] = 'application/json';
+
+        return $this;
+    }
+
+    public function withJsonLdContentType(): self
+    {
+        $this->headers['CONTENT_TYPE'] = 'application/ld+json';
+
+        return $this;
+    }
+
+    public function withMergePatchJsonContentType(): self
+    {
+        $this->headers['CONTENT_TYPE'] = 'application/merge-patch+json';
+
+        return $this;
+    }
+
+    public function withJsonAccept(): self
+    {
+        $this->headers['HTTP_ACCEPT'] = 'application/json';
+
+        return $this;
+    }
+
+    public function withJsonLdAccept(): self
+    {
+        $this->headers['HTTP_ACCEPT'] = 'application/ld+json';
+
+        return $this;
+    }
+
+    public function withShopUserAuthorization(string $email): self
+    {
+        $this->headers = array_merge($this->headers, $this->logInUser('shop', $email));
+
+        return $this;
+    }
+
+    public function withAdminUserAuthorization(string $email): self
+    {
+        $this->headers = array_merge($this->headers, $this->logInUser('admin', $email));
+
+        return $this;
+    }
+
+    /**
+     * @return array <int, string>
+     */
+    public function build(): array
+    {
+        return $this->headers;
+    }
+
+    /**
+     * @return array <string, string>
+     */
+    private function logInUser(string $userType, string $email): array
+    {
+        /** @var UserRepositoryInterface<UserInterface> $shopUserRepository */
+        $shopUserRepository = $this->container->get(sprintf('sylius.repository.%s_user', $userType));
+
+        /** @var JWTTokenManagerInterface $manager */
+        $manager = $this->container->get('lexik_jwt_authentication.jwt_manager');
+
+        /** @var UserInterface|null $user */
+        $user = $shopUserRepository->findOneByEmail($email);
+        Assert::isInstanceOf($user, SymfonyUserInterface::class);
```

Comment:
I think it's outside of scope of this test to check if Sylius User extends Symfony User.


```php
use Sylius\Component\User\Model\UserInterface;
```
```suggestion
        Assert::isInstanceOf($user, UserInterface::class);
```

Reply:
```
phpstan: Parameter #1 $user of method Lexik\Bundle\JWTAuthenticationBundle\Services\JWTTokenManagerInterface::create() expects Symfony\Component\Security\Core\User\UserInterface, Sylius\Component\User\Model\UserInterface given.
```
I couldn't find a more sophisticated way to deal with this static analysis warning 🤷

Reply:
I just used the `instance of` syntax 🕺